core/attributes.c

```diff
@@ -0,0 +1,470 @@
+ * @param uriP
+ * @param tv
+ */
+void lwm2m_updateTransmissionAttributes(lwm2m_attribute_data_t * attributeP, struct timeval *tv)
+{
+    //-------------------------------------------------------------------- JH --
+    if (NULL != attributeP)
+    {
+        attributeP->lastTransmission = tv->tv_sec;
+        if (attributeP->maxPeriod != NULL)
+        {
+            attributeP->nextTransmission = tv->tv_sec + strtol(attributeP->maxPeriod, NULL, 10);
+        }
+        else
+        {
+            attributeP->nextTransmission = 0;
+        }
+    }
+}
+/**
+ * 
+ * @param serverP
+ * @param uriP
+ * @return 
+ */
+lwm2m_attribute_data_t * lwm2m_getAttributes(lwm2m_server_t *serverP, lwm2m_uri_t *uriP)
+{
+    //-------------------------------------------------------------------- JH --
+
+    lwm2m_attribute_data_t * attributeP;
+
+    if (LWM2M_URI_IS_SET_RESOURCE(uriP))
+    {
+        for (attributeP = serverP->attributeData; attributeP != NULL; attributeP = attributeP->next)
+        {
+            if (memcmp(&attributeP->uri, uriP, sizeof(lwm2m_uri_t)) == 0)
+            {
+                return (attributeP);
+            }
+        }
+    }
+    else
+    {
+        for (attributeP = serverP->attributeData; attributeP != NULL; attributeP = attributeP->next)
+        {
+            if (attributeP->uri.objectId == uriP->objectId && attributeP->uri.instanceId == uriP->instanceId)
+            {
+                return (attributeP);
+            }
+        }
+    }
+    return (attributeP);
+}
+
+static uint8_t prv_setAttribute(char** attribute, const void* value, size_t length)
+{
+    if (NULL != *attribute)
+        lwm2m_free(*attribute);
+    *attribute = lwm2m_malloc(length + 1);
+    if (NULL == *attribute)
+        return COAP_500_INTERNAL_SERVER_ERROR ;
+    memcpy(*attribute, value, length);
+    (*attribute)[length] = 0;
+    return COAP_204_CHANGED ;
+}
+
+static uint8_t prv_setAttributeFromString(char** attribute, const char* value)
+{
+    return prv_setAttribute(attribute, value, strlen(value));
+}
+
+/**
+ * Method to validate and set server sent attribute values!
+ * @param contextP LWM2M root context, for to reach other functionality
+ * @param uriP     URI to define related resource
+ * @param attrType type of attribute to set (lt,gt,st,pmin,pmax)
+ * @param value    single attribute value as char*
+ * @param objectP  related object
+ * @param serverP  erver data
+ * @return 
+ */
+uint8_t lwm2m_setAttributes(lwm2m_context_t * contextP, lwm2m_uri_t * uriP, lwm2m_attribute_type_t attrType,
+        const char* value, int length, lwm2m_object_t* objectP, lwm2m_server_t * serverP)
+{
+    //-------------------------------------------------------------------- JH --
+
+    char valStr[50];    //TODO: size enough ? ... malloc?
+    long valLong = 0;
+    // resource attributes are set
+    lwm2m_attribute_data_t * attributeP = NULL;
+
+    if (!LWM2M_URI_IS_SET_RESOURCE(uriP))
+        return COAP_501_NOT_IMPLEMENTED ;
+
```

Comment:
here call the object read callback. It will tell you if the resource exists and its datatype.